build.gradle.kts

```diff
@@ -568,47 +560,10 @@ distributions.shadow {
 
 /* ******************** HiveMQ composite build ******************** */
 
-val deleteOldSpecs by tasks.registering(Delete::class) {
-    projectDir.resolve("specs").walk().forEach {
-        if (!it.isDirectory && it.name.endsWith("OpenAPI-spec.yaml")) {
-            delete(it)
-        }
-    }
-}
-
-if (gradle.includedBuilds.find { it.name == "hivemq-enterprise" } != null) {
-    tasks.register<Copy>("copyHiveMQSpec") {
-        mustRunAfter(deleteOldSpecs)
-        from(hivemqOpenApi.elements.map { it.first().asFile })
-        into("specs")
-    }
-}
-
-if (gradle.includedBuilds.find { it.name == "hivemq-swarm" } != null) {
-    tasks.register<Copy>("copyHiveMQSwarmSpec") {
-        mustRunAfter(deleteOldSpecs)
-        from(swarmOpenApi.elements.map { it.first().asFile })
-        into("specs")
-    }
-}
-
-if (gradle.includedBuilds.find { it.name == "hivemq-swarm" } != null &&
-    gradle.includedBuilds.find { it.name == "hivemq-enterprise" } != null) {
-    tasks.register("updateSpecs") {
-        dependsOn(deleteOldSpecs)
-        dependsOn(tasks.named("copyHiveMQSpec"))
-        dependsOn(tasks.named("copyHiveMQSwarmSpec"))
-
-        doLast {
-            val gradleProperties = projectDir.resolve("gradle.properties")
-            var text = gradleProperties.readText()
-
-            text = text.replace("(?m)^hivemq-api\\.version=.+".toRegex(), "hivemq-api.version=$version")
-            text = text.replace("(?m)^hivemq-swarm-api\\.version=.+".toRegex(), "hivemq-swarm-api.version=$version")
-
-            gradleProperties.writeText(text)
-        }
-    }
+tasks.register<Sync>("updateSpecs") {
+    from(hivemqOpenApi) { rename { "hivemq-openapi.yaml" } }
```

Comment:
Does Gradle still build when these projects are not available?

Reply:
yes, as hivemqOpenApi falls back to the file in the specs folder